### PR TITLE
Dataset search params

### PR DESCRIPTION
**Pull request by A-Derks**

### CODE REVIEW

File: dkist/net/tests/strategies.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def _supported_attr_types():
     attr_types = list(walker.applymm.registry)
     attr_types.remove(object)
     attr_types.remove(AttrAnd)
-    attr_types.remove(a.dkist.BoundingBox)  # TODO: Support BoundingBox
+    #attr_types.remove(a.dkist.BoundingBox)  # TODO: Support BoundingBox
```
Comment (Cadair) [inline dkist/net/tests/strategies.py:31]:
```suggestion
    attr_types.remove(a.dkist.BoundingBox)
```

File: dkist/net/attrs.py [hunk 2]
```diff
@@ -278,3 +279,60 @@ def __init__(self, bottom_left, *, top_right=None, width: u.deg = None,
...
+
+    Parameters
+    ----------
+    spectralmin : `u.Quantity`
+        The minimum value of the average spectral sampling to search between.
+
+    spectralmax : `u.Quantity`
+        The maximum value of the average spectral sampling to search between.
+    """
+    def __init__(self, spectralmin: u.nm, spectralmax: u.nm):
```
Comment (Cadair) [inline dkist/net/attrs.py:296]: I *think* that we want to put [`u.quantity_input`](https://docs.astropy.org/en/latest/api/astropy.units.quantity_input.html?highlight=quantity_input) on these methods, and in this case enable the spectral equivalency:
```suggestion
   u.quantity_input(equivalencies=u.spectral())
    def __init__(self, spectralmin: u.nm, spectralmax: u.nm):
```

This should validate the units against the type annotations, we should add tests for this obviously.

File: dkist/net/attrs.py [hunk 3]
```diff
@@ -278,3 +279,60 @@ def __init__(self, bottom_left, *, top_right=None, width: u.deg = None,
...
+
+    Parameters
+    ----------
+    spatialmin :
+        The minimum value of the average spatial sampling to search between.
+
+    spatialmax :
+        The maximum value of the average spatial sampling to search between.
+    """
+    def __init__(self, spatialmin, spatialmax):
```
Comment (Cadair) [inline dkist/net/attrs.py:315]: I think these should be in arcsec/pixel?

File: dkist/net/attr_walker.py [hunk 4]
```diff
@@ -143,6 +147,20 @@ def _(wlk, attr, params):
     return params.update({'embargoEndDateMin': attr.start.isot,
                           'embargoEndDateMax': attr.end.isot})
 
+@walker.add_applier(SpectralSampling)
+def _(wlk, attr, params):
+    return params.update({'averageDatasetSpectralSamplingMin': attr.min,
+                          'averageDatasetSpectralSamplingMax': attr.max})
```
Comment (Cadair) [inline dkist/net/attr_walker.py:153]: These too will be quantities you will need to cast them back to float with `to_value`.

File: dkist/net/attr_walker.py [hunk 5]
```diff
@@ -143,6 +147,20 @@ def _(wlk, attr, params):
...
+
+@walker.add_applier(SpatialSampling)
+def _(wlk, attr, params):
+    return params.update({'averageDatasetSpatialSamplingMin': attr.min,
+                          'averageDatasetSpatialSamplingMax': attr.max})
```
Comment (Cadair) [inline dkist/net/attr_walker.py:158]: Same here needs `to_value`.

**Review by Cadair — COMMENTED** [review]
Looks like a good start, the results handling in `client.py` also needs updating so we handle these columns in the results table correctly as well.

File: dkist/net/tests/strategies.py (hunk 1 above)
Comment (A-Derks) [reply dkist/net/tests/strategies.py]: Why do we want to remove this from the list?

File: dkist/net/attr_walker.py [hunk 6]
```diff
@@ -74,6 +74,10 @@ def _(wlk, attr, params):
         return params.update({'hasAllStokes': True})
     if attr.value.lower() == "intensity":
         return params.update({'hasAllStokes': False})
+    if attr.value.lower() == "spectral_axis":
+        return params.update({'hasSpectralAxis': True})
+    if attr.value.lower() == "temporal_axis":
+        return params.update({'hasTemporalAxis': True})
```
Comment (Cadair) [inline dkist/net/attr_walker.py:80]: This needs some thought (probably on my part). For a start I am not sure if you can AND two physobs together, where as the union of these two keys makes sense. Also I need to check the VSO data model to see if there is a neater way of doing this.

File: dkist/net/attrs.py [hunk 7]
```diff
@@ -278,3 +279,61 @@ def __init__(self, bottom_left, *, top_right=None, width: u.deg = None,
...
+
+    Parameters
+    ----------
+    spatialmin :
+        The minimum value of the average spatial sampling to search between.
+
+    spatialmax :
+        The maximum value of the average spatial sampling to search between.
+    """
+    def __init__(self, spatialmin: u.arcsec/u.pix, spatialmax: u.arcsec/u.pix):
```
Comment (Cadair) [inline dkist/net/attrs.py:316]: add `u.quantity_input` here as well, but no equivalency is needed this time:
```suggestion
    @u.quantity_input
    def __init__(self, spatialmin: u.arcsec/u.pix, spatialmax: u.arcsec/u.pix):
```

File: dkist/net/attrs.py [hunk 8]
```diff
@@ -278,3 +279,61 @@ def __init__(self, bottom_left, *, top_right=None, width: u.deg = None,
...
+
+    Parameters
+    ----------
+    temporalmin : `u.Quantity`
+        The minimum value of the average temporal sampling to search between.
+
+    temporalmax : `u.Quantity`
+        The maximum value of the average temporal sampling to search between.
+    """
+    def __init__(self, temporalmin: u.s, temporalmax: u.s):
```
Comment (Cadair) [inline dkist/net/attrs.py:335]:
```suggestion
    @u.quantity_input
    def __init__(self, temporalmin: u.s, temporalmax: u.s):
```

File: dkist/net/attr_walker.py [hunk 9]
```diff
@@ -143,6 +147,20 @@ def _(wlk, attr, params):
     return params.update({'embargoEndDateMin': attr.start.isot,
                           'embargoEndDateMax': attr.end.isot})
 
+@walker.add_applier(SpectralSampling)
+def _(wlk, attr, params):
+    return params.update({'averageDatasetSpectralSamplingMin': attr.min.to_value(equivalencies=float),
+                          'averageDatasetSpectralSamplingMax': attr.max.to_value(equivalencies=float)})
```
Comment (Cadair) [inline dkist/net/attr_walker.py:153]: You need to give the target unit here.
```suggestion
    return params.update({'averageDatasetSpectralSamplingMin': attr.min.to_value(u.nm, equivalencies=u.spectral()),
                          'averageDatasetSpectralSamplingMax': attr.max.to_value(u.nm, equivalencies=u.spectral())})

```

File: dkist/net/attr_walker.py [hunk 10]
```diff
@@ -143,6 +147,20 @@ def _(wlk, attr, params):
...
+
+@walker.add_applier(SpatialSampling)
+def _(wlk, attr, params):
+    return params.update({'averageDatasetSpatialSamplingMin': attr.min.to_value(equivalencies=float),
+                          'averageDatasetSpatialSamplingMax': attr.max.to_value(equivalencies=float)})
```
Comment (Cadair) [inline dkist/net/attr_walker.py:158]:
```suggestion
    return params.update({'averageDatasetSpatialSamplingMin': attr.min.to_value(u.arcsec/u.pixel),
                          'averageDatasetSpatialSamplingMax': attr.max.to_value(u.arcsec/u.pixel)})
```

**Review by Cadair — COMMENTED** [review]
The test fails look like a weird interaction between the unit annotations and hypothesis. I will have to dive into that later.

**Pull request closed without merging**